Fix paths so the game actually works :)

Patch suggested by Andrew Sayman and taken from upstream CVS

http://bugs.gentoo.org/show_bug.cgi?id=90263

Index: src/Game.h
===================================================================
RCS file: /cvsroot/crack-attack/crack-attack/src/Game.h,v
retrieving revision 1.18
retrieving revision 1.19
diff -u -r1.18 -r1.19
--- src/Game.h
+++ src/Game.h
@@ -82,7 +82,7 @@
 #    define GC_DATA_DIRECTORY(x)          DATA_DIRECTORY x
 #  endif
 #  define GC_LOCAL_DATA_DIRECTORY        GC_DD "." GC_BINARY GC_DD
-#  define GC_BINARY_LOCATION             DATA_DIRECTORY ".." GC_DD ".."
+#  define GC_BINARY_LOCATION             BINARY_DIRECTORY GC_DD GC_BINARY
 #else
 #  define GC_DATA_DIRECTORY(x)              ".." GC_DD "data" GC_DD x
 #  define GC_LOCAL_DATA_DIRECTORY           ".." GC_DD "localdata" GC_DD
Index: src/gtk-gui/Makefile.am
===================================================================
RCS file: /cvsroot/crack-attack/crack-attack/src/gtk-gui/Makefile.am,v
retrieving revision 1.4
retrieving revision 1.5
diff -u -r1.4 -r1.5
--- src/gtk-gui/Makefile.am
+++ src/gtk-gui/Makefile.am
@@ -5,10 +5,17 @@
   interface.h   callbacks.h   support.h   gui_main.h   persist.h   modeparser.h
 
 AM_CXXFLAGS = -O2
-INCLUDES = -DDATA_DIRECTORY='"${pkgdatadir}/"'	\
-    -DPACKAGE_DATA_DIR=\""$(datadir)"\" \
-	-DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
-  @PACKAGE_CFLAGS@ @DEBUG_INCLUDES@
+
+if WANT_BINRELOC
+  PATH_INCLUDES = 
+else
+  PATH_INCLUDES = -DDATA_DIRECTORY='"${pkgdatadir}/"' \
+      -DPACKAGE_DATA_DIR=\""$(datadir)"\" \
+    -DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
+		-DBINARY_DIRECTORY='"${bindir}"'
+endif
+
+INCLUDES = @PACKAGE_CFLAGS@ @DEBUG_INCLUDES@ ${PATH_INCLUDES}
 
 AM_LDFLAGS = ${all_libraries}
 LDADD = ${X_PRE_LIBS} ${X_LIBS} ${X_EXTRA_LIBS} -lGL -lGLU -lglut -lXi -lX11 -lXmu @PACKAGE_LIBS@
Index: src/Makefile.am
===================================================================
RCS file: /cvsroot/crack-attack/crack-attack/src/Makefile.am,v
retrieving revision 1.9
retrieving revision 1.11
diff -u -r1.9 -r1.11
--- src/Makefile.am
+++ src/Makefile.am
@@ -19,11 +19,12 @@
 AM_CXXFLAGS = -DBR_PTHREAD=0 @DEBUG_CFLAGS@ @AUDIO_CFLAGS@
 
 if WANT_BINRELOC
+  PATH_INCLUDES = 
+else
   PATH_INCLUDES = -DDATA_DIRECTORY='"${pkgdatadir}/"' \
       -DPACKAGE_DATA_DIR=\""$(datadir)"\" \
-    -DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\"
-else
-  PATH_INCLUDES = 
+    -DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
+		-DBINARY_DIRECTORY='"${bindir}"'
 endif
 
 INCLUDES = @PACKAGE_CFLAGS@ @DEBUG_INCLUDES@ ${PATH_INCLUDES} 
